starlette/datastructures.py

```diff
@@ -238,13 +243,15 @@ def __str__(self) -> str:
         return ", ".join(repr(item) for item in self)
 
 
-class ImmutableMultiDict(typing.Mapping):
+class ImmutableMultiDict(typing.Mapping[_KT, _VT_co]):
+    _dict: typing.Dict[_KT, _VT_co]
+
     def __init__(
         self,
         *args: typing.Union[
-            "ImmutableMultiDict",
-            typing.Mapping,
-            typing.List[typing.Tuple[typing.Any, typing.Any]],
+            "ImmutableMultiDict[_KT, _VT_co]",
+            typing.Mapping[_KT, _VT_co],
+            typing.Sequence[typing.Tuple[_KT, _VT_co]],
```

Comment:
Just for reference of why: https://mypy.readthedocs.io/en/stable/common_issues.html#invariance-vs-covariance